vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt

```diff
@@ -97,6 +97,11 @@ class HomeRoomListFragment :
                 is HomeRoomListViewEvents.Done -> Unit
             }
         }
+
+        roomListViewModel.onEach(HomeRoomListViewState::localRoomIds) {
+            // Local rooms should not exist anymore when the room list is shown
+            roomListViewModel.deleteLocalRooms(it)
```

Comment:
I think `HomeRoomListViewModel` should decide this business logic.

Reply:
My problem is that the VM is still alive while the local room is opened, so moving this logic to the VM means that the local room will be deleted too early. I trigger it from the fragment to delete the rooms only when the room list is visible to the user (ie. the local timeline is closed).
It would be nice to have a foreground/background state to decide when a ViewModel should be "alive".
Do you have any suggestions about it?

Reply:
You could add an Action to the ViewModel, triggered in the `onResume` of the Fragment, which delete all the local rooms. So `HomeRoomListViewState.localRoomIds` could even be removed I think.

Also we do not add public API to the ViewModels, please call `handle()` and add an action.

So we will have in the Fragment:

```kollin
    override fun onResume() {
        super.onResume()
        roomListViewModel.handle(HomeRoomListAction.DeleteAllLocalRoom)
    }
```

Reply:
Done https://github.com/vector-im/element-android/pull/7154/commits/8999b40c1af5f6a9e11868f2b1169c487007697c I moved to `onStart` instead of `onResume`